src/java/org/apache/cassandra/cql3/statements/UpdateStatement.java

```diff
@@ -348,4 +366,34 @@ public AuditLogContext getAuditLogContext()
     {
         return new AuditLogContext(AuditLogEntryType.UPDATE, keyspace(), table());
     }
+
+    public static void evaluateConstraintsForRow(Row row)
+    {
+        Iterator<Cell<?>> cellIt = row.cells().iterator();
+        // check constraint for each column
+        while (cellIt.hasNext())
+        {
+            Cell<?> cell = cellIt.next();
+            ColumnMetadata columnMetadata = cell.column();
+            if (!columnMetadata.hasConstraint()
+                || columnMetadata.isComplex()) // complex column is not supported, for now
+                continue;
+
+            ByteBuffer cellData = cell.buffer();
+            evaluateConstraint(columnMetadata, cellData);
+        }
+    }
+
+    public static void evaluateConstraint(ColumnMetadata columnMetadata, ByteBuffer cellData)
+    {
+        if (columnMetadata.hasConstraint())
```

Comment:
@bbotella you are checking it unnecessarily again, of course it has constraints because otherwise you would hit `continue` in `evaluateConstraintsForRow`. Also, this method does not seem to be used anywhere else but here.